mediabridge/main.py

```diff
@@ -1,49 +1,54 @@
 import logging
-import os
+from contextlib import nullcontext
 from datetime import datetime
 
 import typer as typer
 from tqdm.contrib.logging import logging_redirect_tqdm
 
 from mediabridge.data_processing import wiki_to_netflix
+from mediabridge.definitions import OUTPUT_DIR
 
 
 def main(
     verbose: bool = typer.Option(False, "--verbose", "-v"),
     log: bool = False,
-    test: bool = False,
+    full: bool = False,
 ):
     # create output directory if it doesn't exist
-    if not os.path.exists("./out"):
-        logging.warning(" /out directory does not exist, creating...")
-        os.mkdir("./out")
+    if not OUTPUT_DIR.exists():
+        print(
+            f"[WARNING] Output directory does not exist, creating new directory at {OUTPUT_DIR}"
+        )
+        OUTPUT_DIR.mkdir()
 
     if log:
         # log all messages to file
         logging.basicConfig(
             level=logging.DEBUG,
-            filename=f"out/mb_{datetime.now().strftime('%Y%m%d_%H%M%S')}.log",
-            filemode="w",
+            filename=OUTPUT_DIR.joinpath(
+                f"mb_{datetime.now().strftime('%Y%m%d_%H%M%S')}.log"
+            ),
+            filemode="x",
             format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
             datefmt="%H:%M:%S",
         )
+    else:
+        if verbose:
+            level = logging.INFO
+        else:
+            level = logging.WARNING
+        logging.basicConfig(level=level, format="[%(levelname)s] %(message)s")
+
+    # If writing to stdout (log is false), redirect logging to tqdm.write
```

Comment:
I'm not sure I understand this comment. If we're not logging, we need to redirect tqdm? I thought we only need to redirect it if we *are* logging, so that the tqdm prints don't end up in the log. Or, as per your comment, the logging itself gets redirected to `tqdm.write`? How does that work?

Reply:
I guess I'm looking for a comment along the lines of "tqdm would normally do X. If logging is enabled, use this context manager to make it do Y."

Reply:
See other comment, but logging still outputs errors and other important things to stdout unless we tell main to log to a file with the --log arg. So we want to redirect log messages that we intend to go to stdout through tqdm so we don't interrupt the formatting, but we do not want to redirect when we're logging to a file (because then tqdm will just steal the messages away from logging's filewriter).